mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt

```diff
@@ -592,4 +592,82 @@ class StatefulHomeTest {
     rule.onNodeWithText(strings.tournamentsBadgeJoin).assertIsDisplayed()
     rule.onNodeWithText("Tournament 1", ignoreCase = true).assertIsDisplayed()
   }
+
+  @Test
+  fun given_settingsScreen_when_clickingOnPuzzle_then_ItIsOpened() = runTest {
+    val auth = buildAuth { user("email@example.org", "password", "1") }
+    val store = buildStore {
+      collection("users") { document("1", ProfileDocument(solvedPuzzles = listOf("00008"))) }
+    }
+    val assets =
+        FakeAssetManager(
+            "PuzzleId,FEN,Moves,Rating,RatingDeviation,Popularity,NbPlays,Themes,GameUrl\n" +
+                "00008,r6k/pp2r2p/4Rp1Q/3p4/8/1N1P2R1/PqP2bPP/7K b - - 0 24,f2g3 e6e7 b2b1 b3c1 b1c1 h6c1,1852,74,97,1444,crushing hangingPiece long middlegame,https://lichess.org/787zsVup/black#48\n")
```

Comment:
I see that you pass the same asset multiple time, maybe you can create a class that extend `FakeAssetManager` and return this string when `readText` is called. 

You can create different class depending on the expected behaviour.

We have done this for the [speech recognizer](https://github.com/epfl-SDP/android/tree/main/mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/infrastructure/speech)

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/390/commits/51ac3db25c9965fd4f2656f2fe6a204af68a03b1